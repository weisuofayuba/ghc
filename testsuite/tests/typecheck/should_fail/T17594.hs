module T17594 where

id1 :: forall a. a -> a
id1 @a x = x

id2 :: forall a. a -> a
id2 @_ x = x

id3 :: forall a. a -> a
id3 @a (x :: a) = x
