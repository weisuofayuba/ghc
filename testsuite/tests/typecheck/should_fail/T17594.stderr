
T17594.hs:4:1: error:
    • @-binders in functions are not allowed yet
    • In an equation for ‘id1’: id1 @a x = x
      The equation for ‘id1’ has two value arguments,
        but its type ‘a -> a’ has only one

T17594.hs:7:1: error:
    • @-binders in functions are not allowed yet
    • In an equation for ‘id2’: id2 @_ x = x
      The equation for ‘id2’ has two value arguments,
        but its type ‘a -> a’ has only one

T17594.hs:10:1: error:
    • @-binders in functions are not allowed yet
    • In an equation for ‘id3’: id3 @a (x :: a) = x
      The equation for ‘id3’ has two value arguments,
        but its type ‘a -> a’ has only one
